### PR TITLE
what if file contains white spaces

### PR DESCRIPTION
**Pull request by akshayyadav76**
i used this package and it has same issue 
1. if file contains white space it won't share file
2. i don't no way but when i am shearing my PDF files on gmail it say could't attached file...

### CODE REVIEW

**Comment by GAM3RG33K** [comment]
**This seems to be a bug worth debugging. I am not sure I have encountered this before.
However, I am not confident that replacing all `space` with `_` will resolve this issue.**

For example: 

File path:
 `emulated/0/Downloads/a file to share.pdf` 

output: 
`emulated/0/Downloads/a_file_to_share.pdf` 

in this case, the replacing `space` will work just fine if you are changing the name manually, but when there's a `space` in directory path then this logic will fail.

For example: 
File path:
 `emulated/0/documents to share/share immediately/file.pdf` 
output: 
 `emulated/0/documents_to_share/share_immediately/file.pdf`

In this case, replacing `space` will change the original file path to something which may or may not exist, but it will not point to same path. 

Because, as you may know that in storage, `emulated/0/documents to share` is not the same as `emulated/0/documents_to_share`

**Comment by GAM3RG33K** [comment]
@akshayyadav76 
If you can come up with something which does not require modifying the source file path, I'd be love to test that logic out anytime.

**Comment by akshayyadav76** [comment]
your right i haven't notice about the path change..
we can do something like that

` filePath.split("/").last.replaceAll(" ", "_"); `

**Comment by GAM3RG33K** [comment]
Won't the given solution, with modifying the path or file, will require some manual modification in the actual path. I just realized this now. 

Modifying the file name is not different from the issues with directory name modifications. Also if this plugin is used for general purpose apps with storage access, we can not force a user to use names without `space` or change them manually. 

**for ex**:
`a file to share.pdf` & `a_file_to_share.pdf` are both different files.

If your app is a `Notes` app then the user will have to change the file name manually before they can attach the file to a note.

In conclusion, I believe that we should find & resolve the **real issue** which ignores the files with `space` in names, instead of inflicting manual changes by devs or users.

**Comment by akshayyadav76** [comment]
yes, your right at the end i just want to say that 
when i used your plugin i notice that file that contain space in it's name it is not sharing those files otherwise it working fine..

well now i'am using another plugin for that purpose, thanks for your response  i appreciate it... :)

**Comment by GAM3RG33K** [comment]
Thanks for finding out this bug. I'll create an issue to keep this in mind and resolved it when I get time.
